pipeline_dp/dataset_histograms/histograms.py

```diff
@@ -66,9 +77,39 @@ class HistogramType(enum.Enum):
 
 @dataclass
 class Histogram:
-    """Represents a histogram over integers."""
+    """Represents a histogram over numbers.
+
+    Attributes:
+        lower: lower bound of the whole histogram, always included and always
+          equals to 1 for integer histograms.
+        upper: upper bound of the whole histogram, there is no upper for integer
+          histograms (i.e. +inf) and for floating histograms upper exists and is
+          always included.
+    """
     name: HistogramType
     bins: List[FrequencyBin]
+    lower: Union[None, int, float] = field(init=False)
+    upper: Union[None, float] = field(init=False)
+
+    def __post_init__(self):
+        if len(self.bins) == 0:
```

Comment:
Nice to do that in `__post_init__` and nice trick to use `field(init=False)` (I didn't know that)

Reply:
thanks! :)